tmspaymentsystem/src/by/teachmeskills/ps/services/MerchantService.java

```diff
@@ -0,0 +1,165 @@
+                        bankAccountAtomicReference.set(bankAccount);
+                    });
+            writeBankAccountsToTxt();
+        } else {
+            throw new IllegalArgumentException("Incorrect accountNumber.");
+        }
+        return bankAccountAtomicReference.get();
+    }
+
+    public List<BankAccount> getMerchantBankAccounts(String id) throws NoBankAccountsFoundException {
+        Optional<Merchant> o = merchants.stream().filter(m -> m.getId().equals(id)).findFirst();
+        AtomicReference<List<BankAccount>> bankAccounts = new AtomicReference<>();
+        o.ifPresentOrElse(m -> {
+            m.getBankAccounts().sort(Comparator.comparing(BankAccount::getCreatedAt));
+            m.getBankAccounts().sort(Comparator.comparing(BankAccount::getStatus));
+            bankAccounts.set(m.getBankAccounts());
+        }, () -> {
+            try {
+                throw new NoBankAccountsFoundException("Merchant don't have bank accounts.");
+            } catch (NoBankAccountsFoundException e) {
+                System.out.println(e.getMessage());
+            }
+        });
+        return bankAccounts.get();
+    }
+
+    public void updateBankAccount(String accountNum, String newAccountNum) throws
+            IllegalArgumentException, BankAccountNotFoundException {
+        if (!accountNum.matches("[0-9a-zA-Z]{10}}") && !newAccountNum.matches("[0-9a-zA-Z]{10}")) {
+            throw new IllegalArgumentException("Incorrect account number.");
+        }
+        merchants.forEach(m -> {
+            Optional<BankAccount> o = m.getBankAccounts().stream().filter(b -> b.getAccountNumber().equals(accountNum)).findFirst();
+            if (o.isEmpty()) {
+                try {
+                    throw new BankAccountNotFoundException("Not found bank account.");
+                } catch (BankAccountNotFoundException e) {
+                    System.out.println(e.getMessage());
+                }
+            } else {
+                o.get().setAccountNumber(newAccountNum);
+                try {
+                    writeBankAccountsToTxt();
+                } catch (IOException e) {
+                    System.out.println(e.getMessage());
+                }
+            }
+        });
+    }
+
+    public void deleteBankAccount(String id, String accountNum) throws
+            BankAccountNotFoundException, IllegalArgumentException, IOException, MerchantNotFoundException {
+        Merchant m = getMerchantById(id);
+        if (!accountNum.matches("[0-9a-zA-Z]{10}")) {
+            throw new IllegalArgumentException("Incorrect account number.");
+        }
+        Optional<BankAccount> o = m.getBankAccounts().stream().filter(b -> b.getAccountNumber().equals(accountNum)).findFirst();
+        if (o.isEmpty()) {
+            throw new BankAccountNotFoundException("Not found bank account.");
+        } else {
+            m.getBankAccounts().remove(o.get());
+            writeBankAccountsToTxt();
+        }
```

Comment:
o.ifPresentOrElse(ba -> {
   m.getBankAccounts().remove(o.get());
   writeBankAccountsToTxt();
}, () -> throw new BankAccountNotFoundException("Not found bank account."));